modules/nextflow/src/main/groovy/nextflow/processor/TaskBean.groovy

```diff
@@ -133,7 +133,7 @@ class TaskBean implements Serializable, Cloneable {
         this.condaEnv = task.getCondaEnv()
         this.spackEnv = task.getSpackEnv()
         this.moduleNames = task.config.getModule()
-        this.shell = task.config.getShell() ?: BashWrapperBuilder.BASH
+        this.shell = task.config.getShell(task.isContainerEnabled()) ?: BashWrapperBuilder.BASH
```

Comment:
At this point, `task` will always (?) have a `shell` since it comes with the process defaults. I have not toched the "else"-block here since I think it is actually not necessary (and obviously did not cause issues for me).